axum-extra/src/middleware.rs

```diff
@@ -38,7 +38,5 @@ use tower_layer::Identity;
 /// [`HandleErrorLayer`]: axum::error_handling::HandleErrorLayer
 /// [`Infallible`]: std::convert::Infallible
 pub fn option_layer<L>(layer: Option<L>) -> Either<L, Identity> {
-    layer
-        .map(Either::E1)
-        .unwrap_or_else(|| Either::E2(Identity::new()))
+    layer.map_or_else(|| Either::E2(Identity::new()), Either::E1)
```

Comment:
I actually think `map().unwrap_or_else()` is easier to parse. Closures as the first argument is generally uncommon in Rust. So I'd rather keep this as is. Same goes for the other similar changes in this PR.